client/src/Components/HOC/public.jsx

```diff
@@ -0,0 +1,34 @@
+import React, { Component } from "react";
+import { Redirect } from 'react-router-dom';
+import checkLogin from "../helper/checkLogin";
+import Home from '../Layouts/Home';
+
+export default WrappedComponent =>
+  class extends Component {
+    state = {
+      fetch: false,
+      login: null,
+      error: null
+    };
+
+    componentDidMount() {
+        checkLogin()
+        .then(response => {
+          const { result } = response;
+          this.setState({ login: result, fetch: true });
+        })
+        .catch(error => {
+          this.setState({ error, fetch: true });
```

Comment:
you should handle the error correctly 
```suggestion
          this.setState({ error, fetch: true });
```